Capturing technical nuances in Slim CI (#2810)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/cloud-ci-job.md

---

> dbt Cloud makes it easy to test every single code change you make prior to deploying that new logic into production. Once you've connected your [GitHub account](/docs/collaborate/git/connect-github), [GitLab account](/docs/collaborate/git/connect-gitlab), or [Azure DevOps account](/docs/collaborate/git/connect-azure-devops), you can configure continuous integration (CI) jobs to run when someone opens a new pull request in your dbt repository. For more information, refer to [Configuring a webhook](#configuring-a-webhook).

Author: ```suggestion
dbt Cloud makes it easy to test every single code change you make prior to deploying that new logic into production. Once you've connected your [GitHub account](/docs/collaborate/git/connect-github), [GitLab account](/docs/collaborate/git/connect-gitlab), or [Azure DevOps account](/docs/collaborate/git/connect-azure-devops), you can configure continuous integration (CI) jobs to run when someone opens a new pull request in your dbt repository. For more information, refer to [Configuring a job trigger](#configuring-a-job-trigger).
```

> dbt Cloud makes it easy to test every single code change you make prior to deploying that new logic into production. Once you've connected your [GitHub account](/docs/collaborate/git/connect-github), [GitLab account](/docs/collaborate/git/connect-gitlab), or [Azure DevOps account](/docs/collaborate/git/connect-azure-devops), you can configure continuous integration (CI) jobs to run when someone opens a new pull request in your dbt repository. For more information, refer to [Configure a job trigger](#configure-a-job-trigger).

Author: ```suggestion
dbt Cloud makes it easy to test every single code change you make prior to deploying that new logic into production. Once you've connected your [GitHub account](/docs/collaborate/git/connect-github), [GitLab account](/docs/collaborate/git/connect-gitlab), or [Azure DevOps account](/docs/collaborate/git/connect-azure-devops), you can configure continuous integration (CI) jobs to run when someone opens a new pull request in your dbt repository. For more information, refer to [Configure CI for a job](#configure-ci-for-a-job).
```

> dbt Cloud makes it easy to test every single code change you make prior to deploying that new logic into production. Once you've connected your [GitHub account](/docs/collaborate/git/connect-github), [GitLab account](/docs/collaborate/git/connect-gitlab), or [Azure DevOps account](/docs/collaborate/git/connect-azure-devops), you can configure continuous integration (CI) jobs to run when someone opens a new pull request in your dbt repository. For more information, refer to [Configuring a job trigger](#configuring-a-job-trigger).

Author: ```suggestion
dbt Cloud makes it easy to test every single code change you make prior to deploying that new logic into production. Once you've connected your [GitHub account](/docs/collaborate/git/connect-github), [GitLab account](/docs/collaborate/git/connect-gitlab), or [Azure DevOps account](/docs/collaborate/git/connect-azure-devops), you can configure continuous integration (CI) jobs to run when someone opens a new pull request in your dbt repository. For more information, refer to [Configure a job trigger](#configure-a-job-trigger).
```

## Configuring continuous integration in dbt Cloud

> When you set up a dbt Cloud CI job with a [job trigger](#configure-a-job-trigger), dbt Cloud will listen for webhooks from GitHub, GitLab, or Azure DevOps indicating that a new pull request has been opened or updated with new commits. When one of these webhooks is received, dbt Cloud will enqueue a new run of the CI job. 

Author: ```suggestion
When you [set up CI for a job](#configure-ci-for-a-job), dbt Cloud will listen for webhooks from GitHub, GitLab, or Azure DevOps indicating that a new pull request has been opened or updated with new commits. When one of these webhooks is received, dbt Cloud will enqueue a new run of the CI job. 
```

<Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/using_ci_dbt_cloud.png" title="Viewing the temporary schema name for a run triggered by a PR"/>

After completing the dbt run, dbt Cloud will update the pull request in GitHub, GitLab, or Azure DevOps with a status message indicating the results of the run. The status message will state whether the models and tests ran successfully or not. You can enable a setting in your git provider that makes "successful pull request checks" a requirement to merge code. And finally, once the pull request is closed or merged, dbt Cloud will delete the temporary schema from your <Term id="data-warehouse" />.

> dbt Cloud might not drop the temporary schema from your data warehouse if your project has database / schema customization using the [`generate_database_name`](/docs/build/custom-databases#generate_database_name) / [`generate_schema_name`](/docs/build/custom-schemas#how-does-dbt-generate-a-models-schema-name) macros. For more information, refer to [Temp PR schema limitations](/docs/deploy/cloud-ci-job#temp-pr-schema-limitations).

Author: ```suggestion
dbt Cloud might not drop the temporary schema from your data warehouse if your project has database or schema customization using the [`generate_database_name`](/docs/build/custom-databases#generate_database_name) or [`generate_schema_name`](/docs/build/custom-schemas#how-does-dbt-generate-a-models-schema-name) macros. For more information, refer to [Temp PR schema limitations](/docs/deploy/cloud-ci-job#temp-pr-schema-limitations).
```

> ### Configure a job trigger

Author: ```suggestion
### Configure CI for a job
```

> If you want dbt Cloud to run the job whenever a pull request or commit is made, you can set up a job trigger. 

Author: ```suggestion
If you want dbt Cloud to run the job whenever a pull request or commit is made, you can set up continuous integration (CI) for the job. 
```

> To set up a webhook:

Author: ```suggestion
To set up CI:
```

> 1. Navigate to the **Triggers** section of the jobs settings page

Author: ```suggestion
1. Create a new job or edit an existing job to open the settings page.
2. Navigate to the **Triggers** section.
```

> 2. Select **Continuous Integration (CI)**. 

Author: ```suggestion
3. Select **Continuous Integration (CI)**. 
```

> 3. Select **Run on Pull Requests?** as shown in the following image.

Author: ```suggestion
4. Select **Run on Pull Requests?** as shown in the following image.
```

> When you set up a dbt Cloud CI job with a [webhook configured](#configuring-a-webhook), dbt Cloud will listen for webhooks from GitHub, GitLab, or Azure DevOps indicating that a new pull request has been opened or updated with new commits. When one of these webhooks is received, dbt Cloud will enqueue a new run of the CI job. 

Author: ```suggestion
When you set up a dbt Cloud CI job with a [job trigger](#configure-a-job-trigger), dbt Cloud will listen for webhooks from GitHub, GitLab, or Azure DevOps indicating that a new pull request has been opened or updated with new commits. When one of these webhooks is received, dbt Cloud will enqueue a new run of the CI job. 
```

<Lightbox src="/img/docs/dbt-cloud/using-dbt-cloud/using_ci_dbt_cloud.png" title="Viewing the temporary schema name for a run triggered by a PR"/>

After completing the dbt run, dbt Cloud will update the pull request in GitHub, GitLab, or Azure DevOps with a status message indicating the results of the run. The status message will state whether the models and tests ran successfully or not. You can enable a setting in your git provider that makes "successful pull request checks" a requirement to merge code. And finally, once the pull request is closed or merged, dbt Cloud will delete the temporary schema from your <Term id="data-warehouse" />.

dbt Cloud might not drop the temporary schema from your data warehouse if your project has database / schema customization using the [`generate_database_name`](/docs/build/custom-databases#generate_database_name) / [`generate_schema_name`](/docs/build/custom-schemas#how-does-dbt-generate-a-models-schema-name) macros. For more information, refer to [Temp PR schema limitations](/docs/deploy/cloud-ci-job#temp-pr-schema-limitations).

> ### Configure a job trigger

Author: @joellabes I tried to clarify that this is not the newly released webhooks, but a job trigger as seen here (the tab is actually also called "webhook"): 
![Screenshot 2023-02-02 at 1 03 59 PM](https://user-images.githubusercontent.com/3880403/216449811-9b5ad5a0-6a07-4dab-b630-b5ef4da0ce63.png)

Reviewer: Yeah I think you’ve doen a good job threading the needle here!

Reviewer: I've [asked in the webhooks channel](https://dbt-labs.slack.com/archives/C04KR8M25AM/p1675375172516919) if we should rebrand this page

Author: This new UI is now part of this PR thanks to @thorn14

> ### Configuring a webhook

Author: ```suggestion
### Configure a job trigger
```

> If you want dbt Cloud to run the job whenever a pull request or commit is made, you can set up a webhook. 

Author: ```suggestion
If you want dbt Cloud to run the job whenever a pull request or commit is made, you can set up a job trigger. 
```
